pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -651,12 +651,11 @@ func getSizeFromDataVolumeTemplates(vm *k6tv1.VirtualMachine, dataVolumeName str
 		if dvTemplate.Name == dataVolumeName {
 			if dvTemplate.Spec.PVC != nil {
 				return dvTemplate.Spec.PVC.Resources.Requests.Storage()
+			} else if dvTemplate.Spec.Storage != nil {
+				return dvTemplate.Spec.Storage.Resources.Requests.Storage()
```

Comment:
I see that this function is used for defining `pvcSize`, but it's a bit tricky for the Filesystem DVs/PVCs.

The flow where it breaks: my boot source images are stored as VolumeSnapshots with `restoreSize: 30Gi `
In UI, I create a VM with DataVolumeTemplate, and it requests 30Gi, but my target DV/PVC is Filesystem, so it needs a filesystem overhead that 'storage' api gladly provides, and we get this: 

```
$ oc get dv centos-stream10-magenta-meadowlark-63-volume  -ojson | jq .spec.storage.resources.requests.storage
"30Gi"

$ oc get pvc centos-stream10-magenta-meadowlark-63-volume -ojson | jq .spec.resources.requests.storage
"34087042032" <------ about 31.7Gi
```

So probably the best way to get a PVC size is to look at `PVC.spec.resources.requests.storage`

Reply:
but if the data volume is 30gb and the pvc is 50gb the real size of the disk is 30gb no?

Reply:
If the data volume is 30gb, PVC can be 30gb+filesystem overhead (default fs overhead is 0.05)
So PVC size will be bigger than DV requested, but VM will see that it has 30gb only

Reply:
@aglitke WDYT?